### DISCUSSION TITLE
Adapter steigt aus

### DISCUSSION THREAD

**Discussion by chriskloe** [openv/openv#565] [General]
Hallo!

Ich betreibe vcontrold mit vMon an einem openhab 3. Aktuell habe ich gerade http-Authentifizierung in vMon implementiert , damit es überhaupt mit openhab 3 funktioniert (was auch eine lange Lernkurve war).
Jetzt bin ich am Stand, dass vMon die ersten Male länger am Stück durchläuft aber ich habe immer wieder das Problem, dass vcontrold nicht mehr vom Adapter lesen kann und sich danach nicht mehr neu starten lässt. Im dmesg findet sich etwa sowas:

> $ dmesg | grep ttyUSB
> [    9.071670] usb 1-1.5: ch341-uart converter now attached to ttyUSB0
> [ 1731.307724] ch341-uart ttyUSB0: usb_serial_generic_read_bulk_callback - urb stopped: -32
> [ 1731.307861] ch341-uart ttyUSB0: usb_serial_generic_read_bulk_callback - urb stopped: -32
> [ 1731.562677] ch341-uart ttyUSB0: ch341-uart converter now disconnected from ttyUSB0
> [ 1731.900063] usb 1-1.5: ch341-uart converter now attached to ttyUSB1
> [ 2251.503555] ch341-uart ttyUSB1: ch341-uart converter now disconnected from ttyUSB1
> [ 2251.843223] usb 1-1.5: ch341-uart converter now attached to ttyUSB1
> [72727.791804] ch341-uart ttyUSB1: ch341-uart converter now disconnected from ttyUSB1
> [72728.126790] usb 1-1.5: ch341-uart converter now attached to ttyUSB1
> [72849.901936] ch341-uart ttyUSB1: ch341-uart converter now disconnected from ttyUSB1
> [72850.237630] usb 1-1.5: ch341-uart converter now attached to ttyUSB1
> [75417.838861] ch341-uart ttyUSB1: ch341-uart converter now disconnected from ttyUSB1
> [75418.175632] usb 1-1.5: ch341-uart converter now attached to ttyUSB0
> [77730.283956] ch341-uart ttyUSB0: usb_serial_generic_read_bulk_callback - urb stopped: -32
> [77730.284094] ch341-uart ttyUSB0: usb_serial_generic_read_bulk_callback - urb stopped: -32
> [77730.291477] ch341-uart ttyUSB0: ch341-uart converter now disconnected from ttyUSB0
> [77730.630289] usb 1-1.5: ch341-uart converter now attached to ttyUSB1

Zum Einen sehe ich da, dass der Adapter immer wieder von ttyUSB0 auf 1 und zurück springt und er steigt wohl öfter mal aus.
Hat da jemand eine Idee für mich, woran das liegt? Ist da auch ein Problem im vcontrold dahinter, das den Fehler korrekt handhaben müsste um die Verbindung wieder hin zu kriegen (zum Beispiel ttyUSB0 wieder freigeben)?

Auch noch interessant: von meinem Windows-Laptop aus läuft die Übergabe, jedoch wird die Einheit "°C" falsch in der Shell dargestellt. Auf dem Raspi, wo auch vcontrol läuft, erscheint die Ausgabe korrekt, vom Server gibt es aber einen Code 400 (Bad request). Der Aufruf ist identisch, die Ausgabe des Statuscodes habe ich hinzugefügt und bin so der fehlenden Authentifizierung durch ein 401 auf die Schliche gekommen.

Vielen Dank im Voraus für jede Inspiration!

**Comment by Bavarese** [comment]
Hallo,
....sieht doch sehr nach einem ernsthaften Hardwareproblem aus.

Ideen und Fragen dazu:
- welche Adapter-Hardware wird genau verwendet?

- paßt der Treiber zum verwendeten Chip (im Adapter)...?
Zusatz: es sollen viele gefälschte Chips im Umlauf sein. Betrifft aber meines Wissens vorwiegend FTDI-Nachbauten...

- Fremdlicht zu 100 % abgeschirmt (bei optischer Übetragung)... ?  Mußte mit Panzertape nacharbeiten.

- Netzteil prüfen. Ich verwende nur noch die kräftigsten Originalnetzteile von Raspberry.
Mit allen anderen gab es (früher oder später) massive Funktionsstörungen.

- man kann die USB-Geschwindigkeit herabsetzen (auf das Niveau von USB 1). Bitte selbst gugeln  ;-)

Grüsse

**Comment by chriskloe** [comment]
Servus,

sorry für die Verzögerung, ich bin zeitenweise viel auf Reisen.

Der Adapter stammt von ebay und meldet sich per lsusb so:
"Bus 001 Device 056: ID 1a86:7523 QinHeng Electronics CH340 serial converter"
Wie soll ich sagen, ein Chinese sagte in einem Fall mal sowas: "Das da ist ein anderes Cafe" (sah aus wie Starbucks).
Als Treiber habe ich nichts besonderes installiert, das lief beim Rapsbian out of the box.

Den Rest werde ich mal durchprobieren. Jetzt muss ich erstmal meine Anbindung an openhab umstellen. Ich habe eine Lösung gefunden, die in beide Richtungen funktioniert und gleichzeitig hoffentlich das Problem der Zeichenkodierung behebt, das ich zwischen dem Raspi und dem openhab-Server habe.

Vielen Dank soweit!
Viele Grüße,
Christoph

**Comment by chriskloe** [comment]
Seit ich openhab nicht mehr mit vMon anbinde sondern mit vclient, ist der Fehler nicht mehr aufgetreten. Der erste Anlauf der udev-Regel war nur scheinbar nicht korrekt. Mal schaun, wie es weiter läuft.
Auf jeden Fall danke fürs zuhören! ;-)
